### ISSUE TITLE
Release securedrop-workstation-dom0-config-0.8.0

### ISSUE THREAD

**Issue by rocodes** [freedomofpress/securedrop-workstation#866]
This is a tracking issue for the release of the securedrop-workstation-dom0-config-0.8.0 RPM. This release moves the base Fedora template from Fedora 36 to Fedora 37.

## Roles:
- RM: @rocodes 
- Deputy RM: @legoktm 
- Comms: handled by RM/deputy RM
- QA: @cfm, @eloquence (?), others, as needed

### Pre-QA:
- [x] last QA+ merge of [F37 PR](https://github.com/freedomofpress/securedrop-workstation/pull/859)
- [x] Create test plan
- [x] Create release branch

### RC1 QA: 
- [x] [In release branch] bump rc version using `update_version` script, update changelog in rpm .spec file
- [x] [In release branch] create rc tag (signed with a maintainer's key)
- [x] [In release branch] build rc, commit build logs, and publish artifact to yum-test
- [x] QA; if additional RCs needed, repeat this section, amending test plan as needed  

### Release:
- [x] [In release branch] Bump version via `update_version` script, and update [changelog in rpm .spec file](https://github.com/freedomofpress/securedrop-workstation/blob/main/rpm-build/SPECS/securedrop-workstation-dom0-config.spec)
- [x] [In release branch]  Create prod tag (signed by release signing key)
- [x] Build prod artifact, sign with release key, commit build logs 
- [x] QA/smoketest of prod artifact (stuff rpm in dom0 of SDW prod machine)
- [x] Publish prod artifact to yum repo  

### Post-release
- [ ] Backport version and changelog updates to main branch from release branch
- [ ] Update [release docs](https://developers.securedrop.org/en/latest/workstation_release_management.html#release-securedrop-workstation-dom0-config) as needed 
- [x] Retro if/as needed

**Comment by rocodes** [comment]
Test Plan
----------------------
### Clean install scenario (staging package)
Prep: Install Qubes 4.1.2. Provision workstation with `make staging`. Update workstation.
When package is pushed to staging, run updater.
  - [ ] Updater completes successfully
  - [ ] dom0-ws-config package is 0.8.0-rc1
  - [ ] sd-fedora-37-dvm created and assigned to sys-usb
  - [ ] all vms boot / make test in dom0 passes

### Upgrade scenario (staging)
Prep: Existing up-to-date SDW staging setup using fedora-36 vms.
When package is pushed to staging, run updater.
  - [ ] Updater completes successfully
  - [ ] dom0-ws-config package is 0.8.0-rc1
  - [ ] fedora 37 template downloaded
  - [ ] f37, f37-dvm, sd-f37-dvm created and assigned
  - [ ] all vms launch / make test in dom0 passes

**Comment by eloquence** [comment]
### Upgrade scenario (staging)
Prep: Existing up-to-date SDW staging setup using fedora-36 vms.
When package is pushed to staging, run updater.
  - [x] Updater completes successfully
  - [x] dom0-ws-config package is 0.8.0-rc1
  - [x] fedora 37 template downloaded
  - [x] f37, f37-dvm, sd-f37-dvm created and assigned
  - [ ] all vms launch / make test in dom0 passes [not tested yet, will confirm tomorrow]

**Comment by rocodes** [comment]
@eloquence Thank you for testing! For tomorrow, see my updated test plan re test failures on staging; the grsec version we use on staging is later than the one the dom0 tests will expect.

**Comment by rocodes** [comment]
### Clean install scenario (staging package)
Prep: Install Qubes 4.1.2. Provision workstation against 0.8.1-rc1 tag with `make staging`.
  - [x] dom0-ws-config package is 0.8.0-rc1
  - [x] sd-fedora-37-dvm created and assigned to sys-usb
  - [x] all vms boot
  - [x] Updater run completes successfully [Whonix update failed the first round but this is usual flakiness]
  - [ ] make test in dom0 passes: tbd tomorrow

**Comment by eloquence** [comment]
Unfortunately running `make test` uncovered what I think is a release blocker, though one that I hope will be relatively straightforward to resolve.

In the upgrade scenario, `test_sys_usb` failed, specifically this test:
https://github.com/freedomofpress/securedrop-workstation/blob/67a125e22bf09ebf35fed2225049c7e5fdaf2166/tests/test_sys_usb.py#LL11C6-L11C6

This test confirms whether the files needed for USB auto-attachment have been successfully copied. When `sys-usb` is disposable, we have to ensure they are copied into its template, which is now `sd-fedora-37-dvm`.

This is done here:
https://github.com/freedomofpress/securedrop-workstation/blob/main/dom0/sd-workstation.top#L44

That Salt state only gets provisioned during an `sdw-admin --apply` run. The updater doesn't run `sdw-admin --apply` on each run (that would be too slow), it only applies the [`dom0` states](https://github.com/freedomofpress/securedrop-workstation/blob/main/dom0/sd-workstation.top#L5-L21).

In other words, in an upgrade scenario, we end up without the necessary files in `sd-fedora-37-dvm` and thereby in `sys-usb`.

To force the updater to run `sdw-admin --apply`, we have to trigger a full migration. Until we ship the standalone updater (which comes with a better strategy for managing migrations), that's most easily done by placing the migration file in `%post`, as was previously done here:
https://github.com/freedomofpress/securedrop-workstation/pull/700/files#diff-5d2e11f87f29c1289d107de90bd520aa145edde09453d280d1c446e84b7c3aa0R134-R135

**Comment by rocodes** [comment]
@eloquence Thanks for the detailed report - looking at it now

**Comment by legoktm** [comment]
> that's most easily done by placing the migration file in `%post`, as was previously done here:
https://github.com/freedomofpress/securedrop-workstation/pull/700/files#diff-5d2e11f87f29c1289d107de90bd520aa145edde09453d280d1c446e84b7c3aa0R134-R135

I haven't tested it, but I've put what you suggested in PR form at https://github.com/freedomofpress/securedrop-workstation/pull/871 so it's easier to test+discuss.

**Comment by rocodes** [comment]
### RC2 QA
**Upgrade scenario**
Starting with staging sdw setup with fedora-36 as the base system template, run the updater.  (For those whose machines have already been through upgrade testing, you can leave the `fedora-37` template on your machines to avoid the download time. Just remove the sd-fedora-37-dvm and ensure all sd sys vms and the qubes global template settings are set to fedora-36, not 37, before testing).
- [ ] updater completes successfully
- [ ] fedora 37, fedora-37-dvm, and sd-fedora-37-dvm templates present
- [ ] `make test` in dom0 only shows kernel version mismatch failure (expected version is 5.15.98-1-grsec-workstation)

**Comment by eloquence** [comment]
I didn't fully reprovision, but upgrading from RC1 to RC2 resolved the `sys-usb` issue for me, the required files are now present, and tests are passing except for the grsec mismatch. :tada:

**Comment by rocodes** [comment]
**Upgrade scenario**
Starting with staging sdw setup with fedora-36 as the base system template, run the updater.

- [x] updater completes successfully
- [x] fedora 37, fedora-37-dvm, and sd-fedora-37-dvm templates present
- [x] `make test` in dom0 only shows kernel version mismatch failure (expected version is 5.15.98-1-grsec-workstation) [note this instruction requires up to date code pulled into dom0 via `make clone`]

**Comment by rocodes** [comment]
Prod QAers (@nathandyer, me, and anyone else on a prod setup): this is a great time to update and check on the status your prod machines, since we are moving towards building the release. :)

**Comment by rocodes** [comment]
### 0.8.0 (prod) QA 

**Fresh install scenario**
- Hardware: T470
- Setup: Install Qubes 4.1.2, copy rpm into dom0, run `sdw-admin --apply`
- [x] Install completes successfully
- [x] VMs are correctly provisioned

**Prod Upgrade scenario** (Thanks @nathandyer )
- Hardware: ?
- Setup: Install SDW using production installation instructions. Update. Then copy RPM into dom0 and run apply.
- [x] Install completes successfully (required a reboot between installing the 0.8.0 RPM and a successful `apply` run. Noting this for later)
- [x] VMs are correctly provisioned
- [x] Basic acceptance testing/basic use successful  

**Upgrade scenario 2**
- Hardware Thinkpad X1 6Gen
- Setup: Sort-of upgrade (machine was on rc2 staging, flipped to production, sdw-admin --apply, copy rpm into dom0, run apply again)
- [x] Install completes successfully (one stalled `apply` run, reran it and was successful).
- [x] VMs are correctly provisioned
